bin/update_cucumber

```diff
@@ -65,11 +64,12 @@ class {
     /**
      * @phpstan-return TCurrentVersionArray
      */
-    private function findComposerGherkinVersion(string $composerConfigStr): array
+    private function findComposerGherkinVersion(string $composerFile): array
     {
-        $composerConfig = $this->jsonDecodeArray($composerConfigStr);
+        $composerConfig = Behat\Gherkin\Filesystem::readJsonFile($composerFile);
```

Comment:
I think the `$assoc` param has been missed here, so this will be returning `stdClass` rather than `array`?

I don't actually see any uses of `readJsonFile` that are / should be returning anything other than an associative array. I prefer to avoid boolean flags where possible : I'd suggest instead  `public function readJsonFileArray($file): array` which would simplify the interface and avoid the type-checking asserts following the calls.

I don't think we need a general-purpose "decode any JSON as array or stdClass or mixed" method if we're not using it?